plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/ClientMetadata.kt

```diff
@@ -3,16 +3,15 @@
 
 package software.aws.toolkits.jetbrains.services.telemetry
 
-import com.intellij.ide.plugins.PluginManager
 import com.intellij.openapi.application.ApplicationInfo
 import com.intellij.openapi.application.ApplicationNamesInfo
 import com.intellij.openapi.util.SystemInfo
 import software.amazon.awssdk.services.toolkittelemetry.model.AWSProduct
 import software.aws.toolkits.jetbrains.settings.AwsSettings
 
 data class ClientMetadata(
-    val productName: AWSProduct = AWSProduct.AWS_TOOLKIT_FOR_JET_BRAINS,
-    val productVersion: String = PluginManager.getPluginByClass(this::class.java)?.version.toString(),
+    val productName: AWSProduct = PluginResolver.getInstance().product,
+    val productVersion: String = PluginResolver.getInstance().version,
```

Comment:
wouldn't we need to inspect on each call?

Reply:
Addressed in the latest revision - each metric event captures the caller's plugin product/version and these are grouped at time of telemetry publish.